graylog-storage-opensearch3/src/test/java/org/graylog/storage/opensearch3/testing/ClientOS.java

```diff
@@ -439,47 +331,61 @@ private void waitForResult(Callable<Boolean> task) {
-        }, "Unable to retrieve index blocks for index " + index);
-
-        return jsonResult.path(index)
-                .path("settings")
-                .path("index")
-                .path("blocks")
-                .fields()
-                .hasNext();
+        final GetIndexResponse indexResponse = opensearchClient.sync(c -> c.indices().get(req -> req.index(index)), "Failed to get index " + index);
+        return Optional.ofNullable(indexResponse.get(index))
+                .map(org.opensearch.client.opensearch.indices.IndexState::settings)
+                .map(IndexSettings::index)
+                .map(IndexSettings::blocks)
+                .map(blocks ->
+                        Boolean.TRUE.equals(blocks.searchOnly()) ||
+                                Boolean.TRUE.equals(blocks.write()) ||
+                                Boolean.TRUE.equals(blocks.readOnly()) ||
+                                Boolean.TRUE.equals(blocks.readOnlyAllowDelete()) ||
+                                Boolean.TRUE.equals(blocks.metadata()) ||
+                                Boolean.TRUE.equals(blocks.read()))
+                .orElse(false);
     }
 
 
     public String getClusterSetting(String setting) {
-        final ClusterGetSettingsRequest req = new ClusterGetSettingsRequest();
-        final ClusterGetSettingsResponse response = client.execute((c, requestOptions) -> c.cluster().getSettings(req, requestOptions),
-                "Unable to read OS cluster setting: " + setting);
-        return response.getSetting(setting);
+        final GetClusterSettingsResponse settings = opensearchClient.sync(c -> c.cluster().getSettings(r -> r.flatSettings(true)), "Unable to read OS cluster setting: " + setting);
+        return Stream.of(settings.persistent(), settings.transient_(), settings.defaults())
+                .map(s -> s.get(setting))
+                .filter(Objects::nonNull)
+                .findFirst()
+                .map(s -> s.to(String.class))
+                .orElse(null);
     }
 
     @Override
     public void putFieldMapping(String index, String field, String type) {
-        updateMapping(index, Collections.singletonMap("properties", Collections.singletonMap(field, Collections.singletonMap("type", type))));
+        final Property property = dynamicallyCreateProperty(type);
+        opensearchClient.sync(c -> c.indices().putMapping(r -> r.index(index).properties(field, property)), "Failed to update field type");
+    }
+
+    /**
+     * This is an ugly hack because there is no way to dynamically resolve opensearch property type in os3 client
+     *
+     */
+    private Property dynamicallyCreateProperty(String type) {
+        final JsonpMapper mapper = new JacksonJsonpMapper(objectMapper);
+        final JsonParser parser = mapper.jsonProvider().createParser(new StringReader("{\"type\": \"" + type + "\"}"));
+        return Property._DESERIALIZER.deserialize(parser, mapper);
     }
```

Comment:
We tend to put all those ugly hacks in a jail called : `OSSerializationUtils` :)